protocol.go

```diff
@@ -184,6 +187,9 @@ func (p *Conn) checkPrefix() error {
 
 		// Check for a prefix mis-match, quit early
 		if !bytes.Equal(inp, prefix[:i]) {
+			if p.required {
+				return errors.New("Prefix not found")
```

Comment:
Go error convention is to start errors with lowercase -- please update.

Reply:
I'm aware of the Go error convention (and agree with it), but this package doesn't adhere to it.  I'm simply following the standard established in this package.

Reply:
Actually, I guess there isn't a real "standard" in this package.  I just noticed that one error (`ErrInvalidUpstream`) does start with a lowercase.

I'm willing to update this PR as you requested, but I wanted to push back a little since I'm following what most of the package already does.

Let me know what you want me to do with this error message.  Thanks